src/yachtzee/buffs.ts

```diff
@@ -0,0 +1,215 @@
+import { cliExecute, Effect, haveEffect, itemAmount, maximize, myMeat, print, toInt, use } from "kolmafia";
+import { $effect, $item, $items, clamp, get, getActiveEffects, getActiveSongs, have, isSong, Mood, set } from "libram";
+import { acquire } from "../acquire";
+import { withStash } from "../clan";
+import { baseMeat, burnLibrams, turnsToNC } from "../lib";
+import { farmingPotions, mutuallyExclusive, Potion, variableMeatPotionsSetup } from "../potions";
+import { garboValue } from "../session";
+import { executeNextDietStep } from "./diet";
+import { expectedEmbezzlers, pyecAvailable, shrugIrrelevantSongs } from "./lib";
+
+function yachtzeePotionProfits(potion: Potion, yachtzeeTurns: number): number {
+  // If we have an unused PYEC then
+  // 1) If we don't have an effect, +5 to gained effect duration
+  // 2) If we already have an effect, +5 to existing effect duration
+  // This means that the first use of a potion that we don't already have an effect of is more valuable than the next use
+  const PYECOffset = pyecAvailable() ? 5 : 0;
+  const existingOffset = haveEffect(potion.effect()) ? PYECOffset : 0;
+  const extraOffset = PYECOffset - existingOffset;
+  const effectiveYachtzeeTurns = Math.max(
+    Math.min(
+      yachtzeeTurns - haveEffect(potion.effect()) - existingOffset,
+      potion.effectDuration() + extraOffset
+    ),
+    0
+  );
+  const embezzlerTurns = Math.min(
+    expectedEmbezzlers,
+    Math.max(potion.effectDuration() + extraOffset - effectiveYachtzeeTurns, 0)
+  );
+  const barfTurns = Math.max(
+    potion.effectDuration() + extraOffset - effectiveYachtzeeTurns - embezzlerTurns,
+    0
+  );
+  const embezzlerValue = embezzlerTurns > 0 ? potion.gross(embezzlerTurns) : 0;
+  const yachtzeeValue =
+    (effectiveYachtzeeTurns * 2000 * (potion.meatDrop() + 2.5 * potion.familiarWeight())) / 100; // Every 1lbs of lep ~ 2.5% meat drop
```

Comment:
I have been considering for the longest time to add the value of `shrimp cocktail` to this. Even if the mallprice (5k) isn't reflective of what it sells for, it is at the very least worth 5lbs * 20 turns (5 * 0.025 * baseMeat * 20 = 687.5 meat for a base lep) per potion, at a 33% drop rate. Marginal, but not insignificant when it also contributes to 100s of turns yachtzeechains and embezzlers with robort.
(unfortunately the prices still remain high for some reason, so we aren't auto-using them for embezzliers and chains - but they aren't selling so idk maybe somebody needs to do a mall play with them with the stash that we've built up over the thousands of chains we've done)

Reply:
Out of scope of this PR